x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Sample.java

```diff
@@ -109,7 +112,7 @@ protected TypeResolution resolveType() {
         if (childrenResolved() == false) {
             return new TypeResolution("Unresolved children");
         }
-        var typeResolution = isType(field(), dt -> dt != DataType.UNSIGNED_LONG, sourceText(), FIRST, "any type except unsigned_long").and(
+        var typeResolution = isRepresentableExceptCounters(field(), sourceText(), FIRST).and(
```

Comment:
This was technically allowing counters before, but I'm not sure we actually wanted it (?)

Also, it allowed things like date period, which led to `illegal data type [date_period]` 500 errors